airgun/entities/subscription.py

```diff
@@ -48,12 +48,12 @@ def has_manifest(self):
         May be None if user can't verify reliably if manifest is
         uploaded or not due to missing permissions
         """
-        view = self.navigate_to(self, 'All')
+        view = ManageManifestView(self.browser)
```

Comment:
I don't think this will work. To upload manifest in UI, you need to click "Manage Manifest" button, which opens modal window for file upload. When file is selected, Satellite automatically closes modal. And this line would work only if modal is already displayed in browser.

`navigate_to` method is here to ensure that we are on view that we expect to be. If you skip `navigate_to`, you need to ensure browser state through some other mean.

But then, if you o use `navigate_to`, pay close attention to state that you leave Satellite in. Most of navigation code tries to interact with menu, never checking if that's a thing they actually can do. If you leave modal open, I expect that background overlay will cover menu, and as result cause next navigation attempt to fail.

To sum up, what you *probably* need is navigation to modal at beginning of this method, and closing of modal at the end.

Reply:
We can actually bake this behavior into NavigateStep.pre_navigate, so that before any navigation is attempted there is a check whether the modal exists. In integration_tests use of navmazing, we actually recycled the browser instance when this happened, so that we weren't concerned with how to close the modal, we would just start navigation from scratch.